primitives/evm/src/lib.rs

```diff
@@ -56,12 +56,51 @@ pub struct Vicinity {
 	pub origin: H160,
 }
 
+#[derive(Clone, Copy, Eq, PartialEq, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Debug, Serialize, Deserialize))]
+pub struct WeightInfo {
+	pub ref_time_limit: u64,
+	pub proof_size_limit: u64,
```

Comment:
Each limit should be an Option, because in some cases we don't want to track the metric